test/unit/specs/components/Personalization/index.spec.js

```diff
@@ -10,185 +10,94 @@ OF ANY KIND, either express or implied. See the License for the specific languag
-      config,
-      logger,
-      eventManager
-    });
-    personalization.lifecycle.onResponse({ response: first });
-    personalization.lifecycle.onResponse({ response: second });
-
-    const result = personalization.commands.getDecisions.run({ scopes });
+  it("should return an array of decisions in lifecycle::onResponse for provided decisions scopes", () => {
+    const decisionsScopes = ["Foo1", "Foo3"];
 
-    expect(Array.isArray(result)).toBeTrue();
-    expect(result.length).toEqual(2);
-  });
+    response.getPayloadsByType.and.returnValue(SCOPES_FOO1_FOO2_DECISIONS);
 
-  it("expects getDecisions to return the most recent decision for the scope that was overwritten by the last response", () => {
-    const scopes = ["Foo1"];
-    const first = {
-      getPayloadsByType() {
-        return SCOPES_FOO1_FOO2_DECISIONS;
-      }
-    };
-    const second = {
-      getPayloadsByType() {
-        return SCOPES_FOO1_FOO3_DECISIONS;
-      }
-    };
     const personalization = createPersonalization({
       config,
       logger,
       eventManager
     });
 
-    personalization.lifecycle.onResponse({ response: first });
-    personalization.lifecycle.onResponse({ response: second });
-
-    const result = personalization.commands.getDecisions.run({ scopes });
-
-    expect(Array.isArray(result)).toBeTrue();
-    expect(result.length).toEqual(1);
-    expect(result[0].id).toEqual("TNT:ABC:ABC1");
-    expect(result[0].scope).toEqual(scopes[0]);
-  });
-
-  it("expects getDecisions to return empty array when there are no decisions for that specific scope in the storage", () => {
-    const scopes = ["Foo1", "Foo3"];
-    const response = {
-      getPayloadsByType() {
-        return NO_SCOPES_DECISIONS;
-      }
+    const deferred = defer();
+    const onResponse = func => {
+      deferred.resolve(func({ response }));
     };
-    const personalization = createPersonalization({
-      config,
-      logger,
-      eventManager
-    });
-    personalization.lifecycle.onResponse({ response });
-    const result = personalization.commands.getDecisions.run({ scopes });
-
-    expect(Array.isArray(result)).toBeTrue();
-    expect(result.length).toEqual(0);
-  });
 
-  it("expects getDecisions to return empty array if the storage is empty", () => {
-    const scopes = ["Foo1", "Foo3"];
-
-    const response = {
-      getPayloadsByType() {
-        return [];
-      }
-    };
-    const personalization = createPersonalization({
-      config,
-      logger,
-      eventManager
+    personalization.lifecycle.onBeforeEvent({
+      event,
+      decisionsScopes,
+      onResponse
     });
 
-    personalization.lifecycle.onResponse({ response });
-
-    const result = personalization.commands.getDecisions.run({ scopes });
-
-    expect(Array.isArray(result)).toBeTrue();
-    expect(result.length).toEqual(0);
+    return expectAsync(deferred.promise).toBeResolvedTo({
+      decisions: SCOPES_FOO1_FOO2_DECISIONS
+    });
```

Comment:
Nothing asynchronous is actually happening during this process, so there shouldn't be any need for a promise. I think you can re-work this to:

```
let result;
const onResponse = func => {
  result = func({ response });
};

personalization.lifecycle.onBeforeEvent({
  event,
  renderDecisionsEnabled,
  decisionsScopes,
  onResponse
});

expect(result.decisions).toEqual(expectedResponse);
```

I believe this applies to the other test in this file as well.